### ISSUE TITLE
how can I search for files outside of workspace directories?

### ISSUE THREAD

**Issue by arquer** [rioj7/command-variable#46]
This is not really an issue, or maybe it is, I don't know.
I am used to using `${workspaceFolder}`, `${UserHome}` and other VSCode variables in my tasks, however, for some reason, I can't seem to get those to work inside an input variable created with this extension.
For example I have this:

   ```
     {
            "id": "archive_or_local",
            "type": "command",
            "command": "extension.commandvariable.pickStringRemember",
            "args": {
                "description": "Locally or from archive?",
                "options": [ 
                    ["Locally", "locally"],
                    ["From archive", "${pickFile:ArchiveDir}"]
                ],
                "default": "Locally",
                "pickFile": {
                    "ArchiveDir": {
                        "description": "Which directory?",
                        "fromFolder": {"fixed": "<some_directory_outside_of_workspace_dir>/archive/"},
                        "include": "*/**",
                        "showDirs": true,
                        "display": "fileName"
                        }
                }
            }
        },
```

This variable is supposed to return either "locally" or a path to a folder inside a certain directory.
The directory where I want to look for the possible user picks is outside of my workspace directories, so normally, I would get there by doing something like:

`${workspaceFolder}/../path/to/root/archive/dir`

However, if I put that in the `"fixed"` attribute in the `"fromFolder"` argument it does not work...

Am I doing something wrong? or is this just not supposed to work?

Thank you!

**Comment by rioj7** [comment]
@arquer  try v1.46.0, you can now use all variables in the `fromFolder` filepaths

**Comment by arquer** [comment]
I am using v1.46.0... still not working

**Comment by rioj7** [comment]
@arquer What is the value of your `fromFolder` property?

**Comment by arquer** [comment]
I went back to check on it and now it's working, I think it's because I was not using giving a name to the ${workspaceFolder}, if I use ${workspaceFolder:name} then it works fine.
I have noticed however that the "transform" command does not seem to support it. I have this in one of my variables:

```
        {
            "id": "my_id",
            "type": "command",
            "command": "extension.commandvariable.transform",
            "args": {
                "text": "${pickFile:oneBuild}",
                "find": "${workspaceFolder:name}/some/path/",
                "replace": "",
                "pickFile": {
                    "oneBuild": {
                        "description": "Select build",
                        "fromFolder": {"fixed": "${workspaceFolder:name}/some/path/"},
                        "include": "**/*",
                        "showDirs": true,
                        "display": "fileName"
                        }
                    }
            }
        },
```

The goal of the transform is to remove part of the path returned by pickFile (it would be nice to have that as an option in pickFile). However, if I use  "${workspaceFolder:name}/some/path/" in the "find" argument it does not seem to work.

**Comment by rioj7** [comment]
@arquer 

> (it would be nice to have that as an option in pickFile)

That is the same as issue #18, I have made a start with that (some time ago) and will put it back on the remember list. If you have suggestions on the syntax or features...... I will consider a transform/find/replace like arguments

I have looked at all the strings and see if any might be benefit of variable replacement. You might have found one that has skipped my inspection.
